app/screens/participants_list/index.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import ParticipantsList from './participants_list';
+import type {GlobalState} from '@mm-redux/types/store';
+import {connect} from 'react-redux';
+import {getTeammateNameDisplaySetting, getTheme} from '@mm-redux/selectors/entities/preferences';
+import {makeGetProfilesByIdsAndUsernames} from '@mm-redux/selectors/entities/users';
+
+interface ListProps {
+    userIds: string[];
+}
+
+function makeMapStateToProps() {
+    const getProfilesByIdsAndUsernames = makeGetProfilesByIdsAndUsernames();
+
+    return function mapStateToProps(state: GlobalState, ownProps:ListProps) {
+        const allUserIds = ownProps.userIds;
+        return {
+            teammateNameDisplay: getTeammateNameDisplaySetting(state),
+            theme: getTheme(state),
+            userProfiles: getProfilesByIdsAndUsernames(state, {allUserIds, allUsernames: []}) || [],
```

Comment:
this is not really memoizing  as the second parameter is always a new object.. perhaps it would be best to create a new selector?

Reply:
I've added a new selector.